### PR TITLE
Fixes a windows locale bug with the "Everyone" principal

### PR DESCRIPTION
**Pull request by zack-shoylev**

### CODE REVIEW

File: apis/filesystem/src/test/java/org/jclouds/filesystem/strategy/internal/FilesystemStorageStrategyImplTest.java [hunk 1]
```diff
@@ -127,6 +128,20 @@ public void testCreateContainer() {
...
+      TestUtils.directoryExists(TARGET_CONTAINER_NAME, false);
+      result = storageStrategy.createContainer(CONTAINER_NAME);
+      assertTrue(result, "Container not created");
+      TestUtils.directoryExists(TARGET_CONTAINER_NAME, true);
+
+      storageStrategy.setContainerAccess(CONTAINER_NAME, ContainerAccess.PRIVATE);
+      assertEquals(storageStrategy.getContainerAccess(CONTAINER_NAME), ContainerAccess.PRIVATE);
+      storageStrategy.setContainerAccess(CONTAINER_NAME, ContainerAccess.PUBLIC_READ);
+      assertEquals(storageStrategy.getContainerAccess(CONTAINER_NAME), ContainerAccess.PUBLIC_READ);
+   }
```
Comment (demobox) [inline apis/filesystem/src/test/java/org/jclouds/filesystem/strategy/internal/FilesystemStorageStrategyImplTest.java]: Just curious - is this related to the change above, or just backfilling?

File: apis/filesystem/src/main/java/org/jclouds/filesystem/util/Utils.java [hunk 2]
```diff
@@ -100,12 +103,37 @@ public static void delete(File file) throws IOException {
...
+         e.printStackTrace();
+      }
+      try ( InputStreamReader reader = new InputStreamReader(pr.getInputStream(), Charsets.UTF_8)) {
+         return CharStreams.toString(reader).split(" ")[0];
+      } catch (IOException e) {
+         return "Everyone";
+      }
+   }
+
+   public static String WINDOWS_EVERYONE = getWindowsEveryonePrincipalName();
```
Comment (demobox) [inline apis/filesystem/src/main/java/org/jclouds/filesystem/util/Utils.java]: [minor] Can we make this `final`?

File: apis/filesystem/src/main/java/org/jclouds/filesystem/util/Utils.java [hunk 3]
```diff
@@ -100,12 +103,37 @@ public static void delete(File file) throws IOException {
...
+      try {
+         pr.waitFor();
+      } catch (InterruptedException e) {
+         e.printStackTrace();
+      }
+      try ( InputStreamReader reader = new InputStreamReader(pr.getInputStream(), Charsets.UTF_8)) {
+         return CharStreams.toString(reader).split(" ")[0];
+      } catch (IOException e) {
+         return "Everyone";
+      }
```
Comment (demobox) [inline apis/filesystem/src/main/java/org/jclouds/filesystem/util/Utils.java]: Could you explain this a bit? Why do we need the three blocks here? Could this be simplified into one block?

File: apis/filesystem/src/main/java/org/jclouds/filesystem/util/Utils.java (hunk 2 above)
Comment (andrewgaul) [inline apis/filesystem/src/main/java/org/jclouds/filesystem/util/Utils.java]: Can we stub out this code on non-Windows?

File: apis/filesystem/src/main/java/org/jclouds/filesystem/util/Utils.java [hunk 4]
```diff
@@ -100,12 +103,37 @@ public static void delete(File file) throws IOException {
    }
 
    /**
+    * @return Localized name for the "Everyone" Windows principal.
+    */
+   public static String getWindowsEveryonePrincipalName() {
```
Comment (andrewgaul) [inline apis/filesystem/src/main/java/org/jclouds/filesystem/util/Utils.java]: Maybe just make this a `static` block?

File: apis/filesystem/src/main/java/org/jclouds/filesystem/util/Utils.java [hunk 5]
```diff
@@ -100,12 +103,37 @@ public static void delete(File file) throws IOException {
...
+      } catch (IOException e) {
+         return "Everyone";
+      }
+      try {
+         pr.waitFor();
+      } catch (InterruptedException e) {
+         e.printStackTrace();
+      }
+      try ( InputStreamReader reader = new InputStreamReader(pr.getInputStream(), Charsets.UTF_8)) {
+         return CharStreams.toString(reader).split(" ")[0];
```
Comment (andrewgaul) [inline apis/filesystem/src/main/java/org/jclouds/filesystem/util/Utils.java]: Could you simplify this with:

```java
Process process = new ProcessBuilder("whoami /groups").start()
try {
    String line;
    while ((line = reader.readLine ()) != null) {
        if (line.indexOf("\"S-1-1-0\"") != -1) {
            return line.split(" ")[0];
        }
    }
} finally {
    process.destroy();
}
```

File: apis/filesystem/src/test/java/org/jclouds/filesystem/strategy/internal/FilesystemStorageStrategyImplTest.java (hunk 1 above)
Comment (zack-shoylev) [inline apis/filesystem/src/test/java/org/jclouds/filesystem/strategy/internal/FilesystemStorageStrategyImplTest.java]: Adds the missing test

File: apis/filesystem/src/main/java/org/jclouds/filesystem/util/Utils.java (hunk 4 above)
Comment (zack-shoylev) [inline apis/filesystem/src/main/java/org/jclouds/filesystem/util/Utils.java]: I think it looks cleaner this way...?

**Comment by zack-shoylev** [comment]
@andrewgaul @demobox I think I addressed everything but let me know if I missed something.

Also I should note this is related to

https://issues.apache.org/jira/browse/JCLOUDS-1015

File: apis/filesystem/src/main/java/org/jclouds/filesystem/util/Utils.java [hunk 6]
```diff
@@ -100,12 +102,39 @@ public static void delete(File file) throws IOException {
...
+                     }
+                  }
+               }
+            } finally {
+               process.destroy();
+            }
+         } catch (IOException e) {
+         }
+      }
+      return "Everyone";
```
Comment (demobox) [inline apis/filesystem/src/main/java/org/jclouds/filesystem/util/Utils.java]: Fine to merge as-is - just a minor about perhaps commenting this as a fallback and/or default value?

**Comment by zack-shoylev** [comment]
merged

**Pull request closed without merging**